### ISSUE TITLE
Sheriff Alerts configuration has no history or oversight

### ISSUE THREAD

**Issue by CalebRouleau** [catapult-project/catapult#4011]
TL;DR: It would be nice to have the sheriff alert configuration patterns and notification email be part of the codebase rather than be part of an admin config.

https://chromeperf.appspot.com/edit_sheriffs is the page that admins can use to decide which regexes to alert on, who to notify about bugs, and what labels to add to bugs. The patterns can be rather complex.(A)

My team has several times had changes happen to this configuration that have caused problems. 

1. For a while, alerts were completely turned off, so no one knew about regressions. When we when back to turn them on, we couldn't figure out who had turned them off or why.

2. The Performance-Media label was removed from the config somehow recently, and I don't know how that happened. It caused a bunch of bugs filed to not be properly tracked.

Additionally, it is not discoverable for other members of my team since they are not admins. So how we choose which metrics to alert on is magic to them. If I were to switch teams or take a vacation, no one on the team would be able to figure it out.

A. here's an example of the patterns:
`
ChromiumPerf/*/load_library_perf_tests/time_to_load_library/*cdm*
ChromiumPerf/*/media.android.tough_video_cases/buffering_time/*
ChromiumPerf/*/media.android.tough_video_cases/dropped_frame_count/*
ChromiumPerf/*/media.android.tough_video_cases/seek/*
ChromiumPerf/*/media.android.tough_video_cases/time_to_play/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/*energy_sum/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/cpu_time_percentage_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:cc:effective_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:gpu:effective_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:malloc:effective_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:skia:effective_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:v8:effective_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_os:system_memory:java_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_os:system_memory:native_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_os:system_memory:proportional_resident_size_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/story:power_avg/*
ChromiumPerf/*/media.android.tough_video_cases_tbmv2/video_animation:power_avg/*
ChromiumPerf/*/media.desktop/*energy_sum/*
ChromiumPerf/*/media.desktop/cpu_time_percentage_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_chrome:cc:effective_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_chrome:gpu:effective_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_chrome:malloc:effective_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_chrome:skia:effective_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_chrome:v8:effective_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_os:system_memory:java_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_os:system_memory:native_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.desktop/memory:chrome:all_processes:reported_by_os:system_memory:proportional_resident_size_avg/*
ChromiumPerf/*/media.desktop/story:power_avg/*
ChromiumPerf/*/media.desktop/video_animation:power_avg/*
ChromiumPerf/*/media.media_cns_cases/seek/*wifi*
ChromiumPerf/*/media.media_cns_cases/time_to_play/*wifi*
ChromiumPerf/*/media.mobile/*energy_sum/*
ChromiumPerf/*/media.mobile/cpu_time_percentage_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_chrome:cc:effective_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_chrome:gpu:effective_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_chrome:malloc:effective_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_chrome:skia:effective_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_chrome:v8:effective_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_os:system_memory:java_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_os:system_memory:native_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.mobile/memory:chrome:all_processes:reported_by_os:system_memory:proportional_resident_size_avg/*
ChromiumPerf/*/media.mobile/story:power_avg/*
ChromiumPerf/*/media.mobile/video_animation:power_avg/*
ChromiumPerf/*/media.mse_cases/append/*
ChromiumPerf/*/media.mse_cases/mediasource_opening/*post_load
ChromiumPerf/*/media.mse_cases/mediasource_opening/*pre_load
ChromiumPerf/*/media.mse_cases/mediasource_opening/startup_test*
ChromiumPerf/*/media.tough_video_cases/buffering_time/*
ChromiumPerf/*/media.tough_video_cases/dropped_frame_count/*
ChromiumPerf/*/media.tough_video_cases/seek/*
ChromiumPerf/*/media.tough_video_cases/time_to_play/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/*energy_sum/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/cpu_time_percentage_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:cc:effective_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:gpu:effective_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:malloc:effective_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:skia:effective_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_chrome:v8:effective_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_os:system_memory:java_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_os:system_memory:native_heap:proportional_resident_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/memory:chrome:all_processes:reported_by_os:system_memory:proportional_resident_size_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/story:power_avg/*
ChromiumPerf/*/media.tough_video_cases_tbmv2/video_animation:power_avg/*
ChromiumPerf/*/media_perftests/*/*
`

**Comment by anniesullie** [comment]
@simonhatch 

Yeah, the dashboard has grown way bigger than the alerting configuration was initially designed for, and we know we are going to need to improve it at some point, but it's difficult to design around all the existing use cases and large-scale migration needed.

Backing up, how would you ideally want the configuration to work? I know you'd definitely want it visible to your team, with wider access to change it and an audit log, but what would the UI for choosing which benchmarks and metrics to monitor look like? Media is a good example sheriffing rotation that shows one of the big problems: tbm2 telemetry tests have a complex set of metrics that the dashboard doesn't (yet) know about, and then there are c++ benchmarks like media_perftests and load_library_perf_tests don't have the same structure to the metrics they spit out. How could the dashboard show you what's available to monitor in a way that's easy to understand?

**Comment by CalebRouleau** [comment]
I think there are two separate things here:

1. It would be nice if the alerts configuration were part of the codebase, and then we could have commandline tooling instead of dashboard tooling to tell us which metrics would be covered by a given set of Patterns. 

This seems doable because it's just a translation of what's on the website to commandline tools + normal source control system. Of course it would be harder because you would have to migrate other teams as well.


2. Patterns to select metrics to alert on aren't super user-friendly.

This is true, but I don't mind too much. It really isn't that bad. Probably just improving documentation to tell people to go to the "Browse Graphs" section of the dashboard and copy/paste the ChromiumPerf/<botname>/<benchmark_name>/<metric>/<story> from there.

**Comment by CalebRouleau** [comment]
Sorry I can't think of an ideal version of this right now. I don't think a complete overhaul is needed. Just a couple tweaks.

**Comment by benshayden** [comment]
Moved to https://bugs.chromium.org/p/chromium/issues/detail?id=917888
